feat: export type NavLinkRenderProps (#11553)
remix-run/react-router: .changeset/six-pianos-serve.md

---

> "react-router-dom": major

Reviewer: ```suggestion
"react-router-dom": patch
```

This can just be a patch - no new runtime functionality added, just exporting a missing existing type so it's more of a type bugfix IMO

---

> Exported NavLinkRenderProps type. Usefull when typing custom NavLink callback.

Reviewer: ```suggestion
Export `NavLinkRenderProps` type for easier typing of custom `NavLink` callback
```
